### PR TITLE
Use simarm_x64 when targeting arm

### PR DESCRIPTION
**Pull request by liamappelbe**
32-bit binaries are being deprecated on Mac, so when building for arm we need to use Dart's simarm_x64 mode, which builds a 64-bit version of gen_snapshot that can produce 32-bit arm AOT snapshots.

This is a breaking change, because there's a small change needed in flutter/flutter to support this: https://github.com/flutter/flutter/pull/36710

Testing: I built the flutter examples in this mode using Xcode 10 and verified that they work on an iPhone 4S.
Bug: https://github.com/flutter/flutter/issues/22598

### CODE REVIEW

File: tools/gn [hunk 1]
```diff
@@ -196,10 +196,14 @@ def to_gn_args(args):
 
     gn_args['dart_custom_version_for_pub'] = 'flutter'
 
-    # Make sure host_cpu matches the bit width of target_cpu.
+    # Make sure host_cpu matches the bit width of target_cpu. The only exception
+    # is that 32-bit Mac binaries are deprecated, so we switch to simarm_x64.
     target_is_32_bit = gn_args['target_cpu'] == 'arm' or gn_args['target_cpu'] == 'x86'
     if target_is_32_bit:
-      gn_args["host_cpu"] = "x86"
+      if sys.platform == 'darwin' and gn_args['target_cpu'] == 'arm':
```
Comment (chinmaygarde) [inline tools/gn]: I expected `i386` variants of `gen_snapshot` to not be required for Linux either. Is this not the case?

Comment (liamappelbe) [reply tools/gn]: I suppose so, but I haven't tested that case, and I was trying to limit the scope of this change. Is there a reason to switch linux to simarm_x64?

Comment (chinmaygarde) [reply tools/gn]: > Is there a reason to switch linux to simarm_x64?

We maintain a legacy toolchain for this purpose on Linux. This is preventing compiler updates. We were hoping to upgrade the same expecting this set of updates.

Comment (liamappelbe) [reply tools/gn]: Ok. I'll work on testing this on Linux using a 32 bit Android.

Comment (chinmaygarde) [reply tools/gn]: Thanks. I appreciate it. This will really help Flutter simplify its toolchains and fix numbers hacks & workarounds.

Comment (liamappelbe) [reply tools/gn]: I've tested some example apps from my Linux machine to a Moto G4 and it's working.

**Pull request merged**